### PR TITLE
General cleanup

### PR DESCRIPTION
**Pull request by d4n3436**
This PR makes a general cleanup of the codebase, specifically the files that are present in Discord.Net Labs but not in Discord.Net. 

The changes are the following:

- Fix null collections
- Fix typos
- Fix formatting
- Fix `ThreadMember` `Presence` property not being deserialized due to a typo
- Fix unused `RequestOptions` in `GetActiveThreadsAsync`
- Use `Array.Empty` instead of zero-length arrays
- Use more specific exceptions
- Use method group
- Use computed properties
- Use conditional access
- Use `var`
- Use `??` and `??=`
- Use switch expressions
- Use expression body in getters and setters
- Use `Async` suffix
- Remove redundancies
- Remove trailing commas
- Remove unreachable code
- Remove unnecessary usings, whitespace, parentheses, etc
- Other minor changes

This PR doesn't have any breaking changes, except the change of the type of exceptions that are thrown in some parts.

### CODE REVIEW

**Comment by SylveonDeko** [comment]
That's a lot.

File: src/Discord.Net.Core/Entities/Interactions/ApplicationCommandOption.cs [hunk 1]
```diff
@@ -24,13 +21,13 @@ public string Name
             set
             {
                 if (value == null)
-                    throw new ArgumentNullException($"{nameof(Name)} cannot be null!");
+                    throw new ArgumentNullException(nameof(value));
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Interactions/ApplicationCommandOption.cs]: Are you sure this is correct? aren't the parameters `message`, `parameter`

File: src/Discord.Net.Core/Entities/Interactions/Message Components/ComponentBuilder.cs [hunk 2]
```diff
@@ -31,14 +31,14 @@ public List<ActionRowBuilder> ActionRows
             set
             {
                 if (value == null)
-                    throw new ArgumentNullException(paramName: nameof(ActionRows), message: "Cannot set an component builder's components collection to null.");
+                    throw new ArgumentNullException(nameof(value));
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Interactions/Message Components/ComponentBuilder.cs]: Same comment as above

**Review by quinchs — CHANGES_REQUESTED** [review]

File: src/Discord.Net.Core/Entities/Interactions/ApplicationCommandOption.cs (hunk 1 above)
Comment (d4n3436) [reply src/Discord.Net.Core/Entities/Interactions/ApplicationCommandOption.cs]: Yes, the parameter here is actually `value` and not `Name`.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/nameof

The [docs](https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception.-ctor?view=net-5.0#System_ArgumentNullException__ctor_System_String_) also say that the single param of the constructor of `ArgumentNullException` is for the name of the parameter and not for the message.

I can bring back the message to make it clearer.

Comment (quinchs) [reply src/Discord.Net.Core/Entities/Interactions/ApplicationCommandOption.cs]: I would add the error message

Comment (d4n3436) [reply src/Discord.Net.Core/Entities/Interactions/ApplicationCommandOption.cs]: Done.

File: src/Discord.Net.Core/Entities/Interactions/Message Components/ComponentBuilder.cs (hunk 2 above)
Comment (d4n3436) [reply src/Discord.Net.Core/Entities/Interactions/Message Components/ComponentBuilder.cs]: Done

**Pull request merged**